src/pass.h

```diff
@@ -180,7 +180,7 @@ struct PassRunner {
 
   // Adds the default set of optimization passes; this is
   // what -O does.
-  void addDefaultOptimizationPasses();
+  void addDefaultOptimizationPasses(bool withStackIR = true);
```

Comment:
In general we've avoided such APIs. How about something like
```
enum StackIRUsage { NoStackIR, WithStackIR };
void addDefaultOptimizationPasses(StackIRUsage stackIRUsage = WithStackIR)
```
then the caller would be `addDefaultOptimizationPasses(NoStackIR)` which seems clearer

Reply:
(but see my main comment, which might avoid the need for this part of the change I think)

Reply:
Sure, I can make this change but let's figure out the higher-level question first. Searches for "= true)" and "= false)" show a few more places where this pattern could be cleaned up as well.